pkg/dyninst/procsubscribe/procscan/scanner.go

```diff
@@ -219,11 +226,28 @@ func (p *Scanner) Scan() (
 	removed = make([]ProcessID, 0, noLongerLive.Len())
 	noLongerLive.Ascend(func(pid uint32) bool {
 		removed = append(removed, ProcessID(pid))
-		p.live.Delete(pid)
+		p.mu.live.Delete(pid)
 		return true
 	})
 	noLongerLive.Clear(true)
 
+	p.mu.Lock()
+	for _, newProc := range ret {
+		p.mu.live.ReplaceOrInsert(newProc.PID)
+	}
+	p.mu.Unlock()
+
 	p.lastWatermark = nextWatermark
 	return ret, removed, nil
 }
+
+func (p *Scanner) LiveProcesses() []ProcessID {
```

Comment:
Add a comment that this may be called concurrently with `Scan` and the `Scan` must not be called concurrently.